core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1075,6 +1075,24 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED =
+    ConfigBuilder("spark.shuffle.readHostLocalDisk.enabled")
```

Comment:
@attilapiros @cloud-fan @squito @tgravescs @vanzin,
I think we need to mention in migration guide that this option must be disabled for old shuffle service.

Reply:
What about disabling this feature when `spark.shuffle.useOldFetchProtocol` is true (and updating the error message along with the corresponding docs; see below)?

As `spark.shuffle.useOldFetchProtocol` is already exists and documented in one of the migration guide (I do not know why in sql-migration-guide.md although). 

https://github.com/apache/spark/blob/master/docs/sql-migration-guide.md:
> Since Spark 3.0, we use a new protocol for fetching shuffle blocks, for external shuffle service users, we need to upgrade the server correspondingly. Otherwise, we'll get the error message UnsupportedOperationException: Unexpected message: FetchShuffleBlocks. If it is hard to upgrade the shuffle service right now, you can still use the old protocol by setting spark.shuffle.useOldFetchProtocol to true.

I am creating a new jira and PR with this.

Reply:
The new PR is https://github.com/apache/spark/pull/26869.